taxcalc/records.py

```diff
@@ -659,10 +681,60 @@ def read(self, data):
         self.SOIYR = np.repeat(2008, self.dim)
 
     def mutate_imputations(self):
+        """
+        Calculates self._cmbtp_itemizer for this record
+        """
         self._cmbtp_itemizer = imputation(self.e17500, self.e00100,
                 self.e18400, self.e18425, self.e62100, self.e00700,
                 self.e04470, self.e21040, self.e18500, self.e20800)
 
+    def set_vars_for_corp_tax(self):
+        """
+        Calculates the variables used to distribute the corporate income tax
+
+        compensation: float
+            the sum of wages, payroll tax, employer contribution to insurance,
+            and untaxed voluntary contribution to retirement plans
+        dividends: float
+            the individual's dividends received
+        netcapgains: float
+            the individual's net capital gains earnings
+        bonds: float
+            the individual's total bonds (taxable and untaxable)
+        """
+
+        # unsure whether to use in compensation
+        ss_benefits = self.e02400
+
+        # the individual's compensation package
+        # used to calculate share from labor
+        # = sum of wages (e00200 + e00250),
+        # payroll taxes (employer_share_fica? should I add this? how else?),
+        # employer contrib's to insurance (e33420 and archer MSA, e03600),
+        # and untaxed voluntary contrib's to retirement plans
+        # (e07240, not sure if from employer or voluntary)
+
+    # TODO: not sure if we have good data on insurance or retirment
+    # e00250: other dependent income (not sure whether to include)
+        self.compensation = (self.e00200 + self.e00250 + self.e33420 +
+                             self.e03600 + self.e07240)
+
+        # individual's share of the dividends received
+        self.dividends = self.e00600
+
+    # JTC thinks that dividends are enough to measure stock ownership,
+    # but TPC thinks both capgains and dividends should be included
+        # individual's share of the capital gains received
+        # c23650 = long term + short term + both (calculated in CapGains fn)
+        self.netcapgains = self.e23250 + self.e22250 + self.e23660
+
+        # bonds = tax-exempt interest + taxable interest
+        # indidividual's share of total bonds
+        self.bonds = self.e00400 + self.e00300
+
+        # self-employment (E09400) and pass-through income (E02000)
+        self.e_and_p = self.e09400 + self.e02000
+
 
```

Comment:
We might need to add more docstring/comments for this sections.
200 - wages and salaries
250 - other dependent earned income
33420 - employer provided benefits for dependent care
3600 - Archer MSA deduction amount
7240 - retirement saving contribution

600 - ordinary dividends
23250 - net long term capital gain/loss
22250 - net short term gain/loss
23660 - short term plus long term gain/loss

400 - tax exempt interest
300 - taxable interest

9400 - self-employment tax
2000 - rental real estate, royalties, partnerships, s corporations